🚩: Milestone 6: Vet Clinic | feature/create_db_based_on_diagram

Ruthmy opened the pull request:
## 🚩: Milestone 6: Vet Clinic | feature/create_db_based_on_diagram - >>branch<< -

### Hi there! @NitBravoA92 and Ruthmy here. 👋

This is the PR for the milestone 6 of the Vet Clinic project. The goal of this milestone is to implement a database based on the provided diagram.

### Here is a summary of what has been done:

- Created the database based on the provided diagram for a clinic.
- Created the file `schema_based_on_diagram.sql` with the necesary queries to create the database.
- Created the folowing tables:
    - `patients`.
    - `medical_histories`.
    - `invoices`.
    - `invoice_items`.
    - `treatments`.
- Created the join table `medical_histories_treatments` to manage the many to many relationship between `medical_histories` and `treatments`.

---

Screenshot of the original database:
<img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/26df9f7c4a6479ef67f7a682a58b06d1134292bf/assests/6-task/clinic_diagram.png">
<br>

Screenshot of the database created:
<img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/26df9f7c4a6479ef67f7a682a58b06d1134292bf/assests/6-task/database-from-diagram.png">
<br>

List of relations
<img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/deca8da7d29bc3659f77583067759d21cb796fab/assests/6-task/Lists-of-relations.jpg">
<br>

Index creation:
<img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/ef99080d659309993feadc1df19669fc66a6971a/assests/6-task/create_indexes.png">
<br>

<table>
  <tr>
    <th colspan="2">Screenshots of the tables created</th>
  </tr>
  <tr>
    <td><img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/ef99080d659309993feadc1df19669fc66a6971a/assests/6-task/invoice_items_table.png"></td>
    <td><img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/ef99080d659309993feadc1df19669fc66a6971a/assests/6-task/invoices_table.png"></td>
  </tr>
  <tr>
    <td><img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/ef99080d659309993feadc1df19669fc66a6971a/assests/6-task/medical_histories_table.png"></td>
    <td><img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/ef99080d659309993feadc1df19669fc66a6971a/assests/6-task/medical_histories_treatments_table.png"></td>
  </tr>
  <tr>
    <td><img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/ef99080d659309993feadc1df19669fc66a6971a/assests/6-task/patients_table.png"></td>
    <td><img src="https://raw.githubusercontent.com/Ruthmy/vet-clinic-db/ef99080d659309993feadc1df19669fc66a6971a/assests/6-task/treatments_table.png"></td>
  </tr>
</table>


---

### Thank you for taking the time to review this PR ⭐!

---

NitBravoA92 committed:
Create the patients table

NitBravoA92 committed:
Create the medical_histories table

NitBravoA92 committed:
CReate the invoices table

Ruthmy committed:
Modified 'schema_based_on_driagram'
- Added the tables 'invoice_items', 'treatments'
- Created the join table 'medical_histories_treatments'

Ruthmy committed:
Added the images of the original diagram and the final one.

Ruthmy committed:
Modified 'schema_based_on_diagram.sql'
- Change order of the tables
- Fixed a missing comma

Ruthmy committed:
Added a screenshot of the DB relations

Ruthmy committed:
Modified 'README'
- Deleted an unnecesary commet

AbelG101 commented on schema_based_on_diagram.sql:
- [x] It appears that we missed indexing the foreign keys in this schema. As per the project requirements, it is necessary to index all foreign keys for optimization purposes. Let's make sure we address this oversight.
- [x] Additionally, please remember to index the foreign keys in the "Join table" that you have created. This will help improve the performance of queries involving the relationship between the tables.

Ruthmy replied:
Hola @AbelG101 
We check our code, and you're right, the FK are in there, but we miss adding additional indexes for the tables. 
We are working on it, thank you very much.
Happy codding! ✨

AbelG101 reviewed (changes requested):
![](https://media.giphy.com/media/v1.Y2lkPTc5MGI3NjExNzFmY2U0ZTkwMmM3ZTM5MDBkNjhhZjk0MjI0OTU3MGNjYjMyMmRhOCZjdD1n/M1fddM48g9084bJMEX/giphy.gif)


# Hi @Ruthmy & @NitBravoA92 👋, 

You're doing a good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Join table has been properly created ✔
- Primary keys and foreign keys are correctly set ✔
- Correct use of Git Flow ✔
- Descriptive PR title with a top notch summary ✔

## Required Changes ♻️
- _Check the comments under the review._
- you just have to make a few changes and you're on your way 🚀.


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Ruthmy committed:
Modified 'schema_based_on_diagram.sql'
- Created index for the tables 'treatments' and 'invoice_items'
- Created index for the join table 'medical_histories_treatments'

NitBravoA92 committed:
Create an index in the patient_id column of the medical_histories table

NitBravoA92 committed:
Create an index for the medical_history_id column of the medical_histories table

NitBravoA92 committed:
Add some screenshots of the tables structures

DeliceLydia reviewed (approved):
## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Created indexes on all FOREIGN KEYS



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Ruthmy commented:
Hi @DeliceLydia 
Thank you very much for your review ✨
